bcipy/language/model/gpt2.py

```diff
@@ -132,33 +289,26 @@ def predict(self, evidence: List[str]) -> List[Tuple]:
         Response:
             A list of symbols with probability
         """
+
         assert self.model is not None, "language model does not exist!"
 
         evidence_str = "".join(evidence)
 
-        # mark the start-of-word indicator
-        if evidence_str == "" or evidence[-1] == SPACE_CHAR:
-            self.is_start_of_word = True
-        else:
-            self.is_start_of_word = False
-
-        # predict the next character
         evidence_str = evidence_str.replace(SPACE_CHAR, ' ')
+        evidence_str = evidence_str.lower()
 
-        # if we are at the start of a word, let the language model predict the next word
-        # then we get the probability distribution of the first character of the next word
-        if self.is_start_of_word is True:
-            self.curr_word_predicted_prob = self.__model_infer(evidence_str)
-            next_char_pred = self.__get_char_predictions("")
+        # run beam search
+        all_beam_candidates = self.__beam_search(evidence_str, self.beam_width, self.search_depth)
 
-        # if we are at the middle of a word, just get the marginal distribution of
-        # the next character given current word prefix
-        else:
-            # first we need the language model predict the probability distribution of the
-            # current word
-            self.curr_word_predicted_prob = self.__model_infer(" ".join(evidence_str.split()[:-1]))
-            cur_word_prefix = evidence_str.split()[-1]
-            next_char_pred = self.__get_char_predictions(cur_word_prefix)
+        # get marginalized characger-level probabilities from beam search results
+        next_char_pred = self.__get_char_predictions_beam_search(evidence_str, all_beam_candidates)
+
+        # interpolate with unigram language model to smooth the probability distribution returned
+        # by GPT2 language model
+        next_char_pred = GPT2LanguageModel.interpolate_language_models(dict(next_char_pred), self.unigram_lm, 0.8)
```

Comment:
bring hardcoded coeffecients up to init for easier configuration

Reply:
Yes, I have done it.